### ISSUE TITLE
Concordance and setup: MAFS

### ISSUE THREAD

**Issue by lbestock** [arch-kiosk/arch-kiosk-office#3017]
Brandeis University and St. Andrews University run a project going into its second year. They would like to use Kiosk though they are not 100% certain they can afford the iPads yet. After a two hour lunch with the director I think it is worth setting up a server for them. Lutz always wanted a Scottish project....

Monastic Archaeology Field School

- [x] new server: MAFS
- [x] request field recording from 1st season from Dar
- [x] meet with Dar for concordance

**Comment by urapadmin** [comment]
why not use the demo server first?
Setting up a server is work and leads to maintenance.

**Comment by urapadmin** [comment]
@lbestock What's next here?

**Comment by lbestock** [comment]
I told Dar we were putting her on the listserv, and we agreed we will meet to run through the demo recording in a couple of weeks when the semester settles. But we did not set a date yet and I do not have recording from her.

**Comment by urapadmin** [comment]
okay. I'll point MAFS to the demo server after I have updated it in the next rollout

**Comment by lbestock** [comment]
Dar has been in touch explicitly requesting a walkthrough and we are scheduling a meeting possibly for next week. mafs.arch-kiosk.brown.edu doesn't do anything yet and I thought this meant it would go to the demo? But ah, the demo is down, too...

**Comment by urapadmin** [comment]
I'll prepare it soon.

**Comment by urapadmin** [comment]
warecode or su pottery?

**Comment by lbestock** [comment]
warecode, please

**Comment by lbestock** [comment]
Meeting date is February 21st.

**Comment by urapadmin** [comment]
we should keep in mind that at least theoretically it might be possible to run this online. It sounds small enough and Scotland has the internet.

**Comment by urapadmin** [comment]
What is the terminology here? Currently Kiosk (not the recording app) uses:

```
    locus: 'context'
    unit: 'area'
    locus_photo: 'context photos'
    dayplan: 'area photos'
    locus_relations: 'context relations'
    collected_material_photo: 'collected material & special finds'
    collected_material: 'collected material & special finds'
    unit_narrative: 'area narrative'
    site_note_photo: 'site note'
    site: 'site map'
    pottery_images: 'pottery'
    tags: 'context tags'

```

**Comment by lbestock** [comment]
She says the internet is terrible and they want a local server.
I don't have the recording yet so I don't know terminology.
It's going to be unfun to configure - they use site-wide rather than trench bound CM numbers, and they will run six trenches at once so that's a lot of identifier lists. I laid the groundwork for trying to convince her to use tench-bound context numbers but it seemed like a sticking point.

**Comment by urapadmin** [comment]
the server is at least available now. The recording app is not configured at all, yet. So it is even missing most identification methods and locus types etc.

**Comment by urapadmin** [comment]
as for the admin passwort: https://wiki.arch-kiosk.brown.edu/urapdev/doku.php?id=devices_and_users#mafsarch-kioskbrownedu

hint: not hot but wet

**Comment by lbestock** [comment]
Field manual (which is pretty great) in the drive. Page 8-9 for initial configuration.
Next Dar meeting: 5 May.

**Comment by lbestock** [comment]
Dar flies out the 27th of May.
She WOULD like a local server, has access to FM through Brandeis, and has just received the list of stuff to buy.
She will have 1 week before her team arrives to play with this in Scotland and time to learn before she goes.
We can both meet on the 16th of May for a physical handoff.
12 May 25 zoom meeting to check configuration is final before moving offline.

**Comment by urapadmin** [comment]
MAFS wants lots dependent on the unit or lot. Both are possible:

**lot_domain:** This is the important one here - "unit" makes the lot contingent on the unit, "locus", which is what we recommend, makes it sub-ordinate to the locus.
**lot_auto_generate** has to be yup to auto generate at all (otherwise free input
**lot_input_type should** not be "free input" unless you want exactly that. The combination of free_input here and lot_auto_generate set to yup leads to generating a new lot but giving the user the option to change it before actual creation.
**lot_max_number** caps the lots somewhere
**lot_check_type** "error" makes creating a lot fail after the last lot is being used. Otherwise the user can proceed if the max_number is violated.

**Comment by urapadmin** [comment]
MAFS wants to take collected material numbers from a numbering list that depends on the unit. While that is technically possible it does not at all protect from duplicate numbers (which is what identifier lists were invented for):
iPadA goes to trench A, create A-001. iPadA leaves.
iPadB comes in to A and creates A-001 again because it cannot know about the number iPadA created before synchronization.

So when it comes to protection from duplicate numbers this is not safer than other automatic identification methods that don't use a list. We do this here only to allow partitioning of the numbers on unit level. On Uronarti we see that the collected material CC-001-1 belongs to unit CC because that's in the identifier. MAFS will know because all collected material numbers from 000 to 200 belong to unit CC. 

In other words: This is only safe as long as only one iPad is operating in the same trench per session (between sync rounds).

this is the configuration:
```
generation:
idgen_group
domain=unit::arch_context
list=cm numbers per unit
match_partition=unit::arch_context

formatting:
idfmt_concat
prefix=-
type=numeric
digits=3
```

**Comment by urapadmin** [comment]
@lbestock 
I have set the identification methods up on the basis of the rliim identification methods and a new MAFS method for MAFS collected material. That's the one taking the unit-based number list into account. It is still including unit and locus into the identifier but that could be changed to only using the number or whatever is requested here.

The other archaeological identifiers are currently created according to urap standard.

Lots are activated and they auto generate on the basis of the locus number right now, Can be switched to unit (see above)

**Comment by lbestock** [comment]
I am 99% done configuring this I think, and testing is now smooth. (For creation of records; I have not tried synchronizing any data yet, just config.)

But. In re the above. The locus number is not currently part of the CM number as generated. The trench number is, but not the locus number. Since you told me to expect it I make a comment here, but I haven’t actually looked myself at the ID method.

**Comment by urapadmin** [comment]
> I am 99% done configuring this I think, and testing is now smooth. (For creation of records; I have not tried synchronizing any data yet, just config.)
> 
> But. In re the above. The locus number is not currently part of the CM number as generated. The trench number is, but not the locus number. Since you told me to expect it I make a comment here, but I haven’t actually looked myself at the ID method.

I'll look at that. Is the locus number supposed to be part of the cm number?

**Comment by lbestock** [comment]
She didn’t say so - I was going off what you put here and was more concerned if something you wrote didn’t work the way you anticipated it would.

In fact she might want nothing but the running number in there, I think that’s how they had them before. I’ll check that with her tomorrow.

**Comment by urapadmin** [comment]
I checked. It is indeed currently set to have unit - cm number. Was an accident. But I left it that way and you can talk to her about it.

**Comment by luizaogs** [comment]
This can also be closed?
